ocpp-transport-soap/src/main/kotlin/com/izivia/ocpp/http/HttpClient.kt

```diff
@@ -0,0 +1,111 @@
+package com.izivia.ocpp.http
+
+import com.izivia.ocpp.soap.OcppSoapParser
+import com.izivia.ocpp.soap.RequestSoapMessage
+import com.izivia.ocpp.soap.ResponseSoapMessage
+import com.izivia.ocpp.transport.ClientTransport
+import com.izivia.ocpp.transport.OcppCallErrorException
+import org.http4k.client.JavaHttpClient
+import org.http4k.contract.contract
+import org.http4k.contract.div
+import org.http4k.core.*
+import org.http4k.lens.Path
+import org.http4k.server.Http4kServer
+import org.http4k.server.Undertow
+import org.http4k.server.asServer
+import org.slf4j.LoggerFactory
+import java.util.*
+import kotlin.reflect.KClass
+
+class HttpClient(
+    private val path: String,
+    private val port: Int,
+    private val ocppId: String,
+    private val target: String,
+    private val ocppSoapParser: OcppSoapParser,
+    private val newMessageId: () -> String = { UUID.randomUUID().toString() },
+) : ClientTransport {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(HttpClient::class.java)
+    }
+
+    private val server: Http4kServer
+    private val client = JavaHttpClient()
+
+    private val handlers = mutableListOf<(HttpMessage) -> HttpMessage?>()
+
+    init {
+        val route = path / Path.of("action") / Path.of("ocppId") bindContract Method.POST to ::routeHandler
+        val app = contract {
+            routes += route
+        }
+        server = app.asServer(Undertow(port = port))
+    }
+
+    private fun routeHandler(action: String, ocppId: String): HttpHandler = { request: Request ->
+        val message = HttpMessage(
+            msgId = ocppId,
```

Comment:
it sounds strange that we use the ocpp id as msgId.  Either msgId is bad name, or we should use something else.
and BTW, in the client we deal with a single ocppid, which is given in the constructor, so I don't know why we have it in the route? And if we keep it, we should check it's the same as the configured one